prowler/providers/azure/services/defender/defender_service.py

```diff
@@ -57,12 +58,37 @@ def __get_auto_provisioning_settings__(self):
                         }
                     )
             except Exception as error:
-                logger.error(f"Subscription name: {subscription}")
+                logger.error(f"Subscription name: {subscription_name}")
                 logger.error(
                     f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
                 )
         return auto_provisioning
 
+    def __get_assessments__(self):
+        logger.info("Defender - Getting assessments...")
+        assessments = {}
+        for subscription_name, client in self.clients.items():
+            try:
+                assessments_list = client.assessments.list(
+                    f"subscriptions/{self.subscriptions[subscription_name]}"
```

Comment:
Would not it be `f"subscriptions/{subscription_name}"`?

Reply:
No, because I need the subscription ID no the subscription name. Inside the object there is a dictionary name "subscriptions" that contains:

{ subscription_name: subscription_id}

Reply:
Got it, thanks for the explanation.